src/python/pants/backend/docker/target_types.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, Sources, StringField, Target
+
+
+class DockerImageSources(Sources):
+    default = ("Dockerfile",)
+    expected_num_files = 1
+    help = "Name of the Dockerfile to use when building the Docker image."
+
+
+class DockerImageVersion(StringField):
+    alias = "version"
+    default = "latest"
+    help = "Image tag to apply to built images."
+
+
+class DockerDependencies(Dependencies):
```

Comment:
think we could just use the `Dependencies` field directly. The reason to subclass is usually to override `supports_transitive_excludes`, which you're not doing here. 

Although now that I think of it, maybe we do want to do that here? We do for other non-library targets. I'm not sure of the semantics of `supports_transitive_excludes" and why we don't always set it, so I will defer to @Eric-Arellano .

Reply:
It'll be needed for the dependency inference.

Reply:
Got it.

Reply:
Yeah, I think this makes sense to set `supports_transitive_excludes=True`. 

The idea is that we only enable that for targets that are not usually dependencies of other targets: the roots of your build. It would get really confusing if you had some intermediate `python_library` with a transitive exclude that was causing a bunch of issues later like 

> Why is my `pex_binary` not including this target even when I hardcode it???